pkg/components/yql_agent.go

```diff
@@ -73,6 +75,18 @@ func NewYQLAgent(cfgen *ytconfig.Generator, ytsaurus *apiproxy.Ytsaurus, master
 			getTolerationsWithDefault(resource.Spec.YQLAgents.Tolerations, resource.Spec.Tolerations),
 			getNodeSelectorWithDefault(resource.Spec.YQLAgents.NodeSelector, resource.Spec.NodeSelector),
 		),
+		updateEnvironment: NewInitJob(
+			&l,
+			ytsaurus.APIProxy(),
+			ytsaurus,
+			resource.Spec.ImagePullSecrets,
+			"yql-agent-update-environment",
+			consts.ClientConfigFileName,
+			getImageWithDefault(resource.Spec.YQLAgents.Image, resource.Spec.CoreImage),
```

Comment:
Taking CoreImage as QT/Yqla image is not good. They have initial version there and it will only lead to problems (and we had them already)

Do we take `CoreImage` as default to QueryTracker too?

If yes, then I am fine with proceeding and creating a separate issue for it.

Reply:
Yes, we have it everywhere
Like this in init jobs:
https://github.com/ytsaurus/ytsaurus-k8s-operator/blob/main/pkg/components/query_tracker.go#L82

And as a component image too
https://github.com/ytsaurus/ytsaurus-k8s-operator/blob/main/pkg/components/server.go#L96

Reply:
Since we are only using yt cli in job it should be fine.